TeamCode/src/main/java/org/firstinspires/ftc/teamcode/commands/groups/ZeroRing.java

```diff
@@ -32,54 +33,55 @@ public ZeroRing(MecanumDriveSubsystem drive, WobbleSubsystem wobbleSystem, Shoot
+        Vector2d shootPose = traj1.end().vec().plus(new Vector2d(-25.0, 22.0));
 
         Trajectory traj2 = drive.trajectoryBuilder(traj1.end(), true)
                 .lineToConstantHeading(shootPose)
                 .build();
 
-        Vector2d secondWobble = traj2.end().vec().plus(new Vector2d(-12.0, 12.0));
+        //-34 , -24
+//        Vector2d secondWobble = traj2.end().vec().plus(new Vector2d(-14.0, 15.5));
 
         Trajectory traj3 = drive.trajectoryBuilder(traj2.end(), 0.0)
-                .splineToLinearHeading(new Pose2d(secondWobble, 0.0), Math.toRadians(-90.0))
+                .splineToLinearHeading(new Pose2d(-34.0,-24.0, 0.0), Math.toRadians(-90.0))
                 .build();
 
-        Trajectory traj4 = drive.trajectoryBuilder(traj3.end(), false)
-                .splineToConstantHeading(traj3.end().vec().plus(new Vector2d(-2.2, 8.0)), 0.0)
+        Trajectory traj4 = drive.trajectoryBuilder(traj3.end(), 0.0)
+                .splineToSplineHeading(traj3.end().plus(new Pose2d(16.0, 0.0, Math.toRadians(180.0))), 0.0)
+                .splineToConstantHeading(traj1.end().vec().plus(new Vector2d(-12.0, -3.5)), 0.0)
                 .build();
 
-        Trajectory traj5 = drive.trajectoryBuilder(traj4.end(), 0)
-                .splineTo(traj4.end().vec().plus(new Vector2d(16.0, 0.0)), 0.0)
-                .splineToLinearHeading(traj1.end().plus(new Pose2d(-15.0, -12.0, Math.toRadians(180.0))), 0.0)
+        Trajectory traj5 = drive.trajectoryBuilder(traj4.end())
+                .splineToConstantHeading(traj4.end().vec().plus(new Vector2d(-10.0, 16.0)), 0.0)
+//                .splineTo(traj4.end().vec().plus(new Vector2d(-16.0, 16.0)), 0.0)
+                .splineTo(traj4.end().vec().plus(new Vector2d(25.0, 20.0)), 0.0)
                 .build();
 
         addCommands(
                 new InstantCommand(shooter::shoot, shooter),
-                new TrajectoryFollowerCommand(drive, traj0),
+                new ParallelDeadlineGroup(
+                        new TrajectoryFollowerCommand(drive, traj0),
+                        new Com_PutDown(wobbleSystem)
+                ),
                 new TrajectoryFollowerCommand(drive, traj1),
-                new Com_PutDown(wobbleSystem),
                 new InstantCommand(wobbleSystem::openGrabber, wobbleSystem),
                 new WaitCommand(500),
                 new Com_PickUp(wobbleSystem),
                 new TrajectoryFollowerCommand(drive, traj2),
                 new TurnCommand(drive, Math.toRadians(10)),
                 new RapidFireCommand(shooter),
-                new ParallelDeadlineGroup(
-                        new Com_PutDown(wobbleSystem),
-                        new TurnCommand(drive, Math.toRadians(-10))
-                ),
+                new Com_PutDown(wobbleSystem),
                 new InstantCommand(shooter::stop, shooter),
                 new TrajectoryFollowerCommand(drive, traj3),
```

Comment:
These could probably be in a `ParallelDeadlineGroup`, with the trajectory follower command as the deadline (first command in the group).

Reply:
i.e.
```java
new InstantCommand(shooter::stop, shooter),
new ParalellDeadlineGroup(
  new TrajectoryFollowerCommand(drive, traj3),
  new Com_PutDown(wobbleSystem)
),
```